File: web/src/api/tool.ts
import request from '@/utils/request'

export interface OutputFieldDef {
  type: 'string' | 'number' | 'boolean' | 'object' | 'array'
  label: string
  children?: Record<string, OutputFieldDef>
}

export interface Tool {
  id?: string
  code: string
  name: string
  description: string
  category: string
  version: string
  author: string
  icon: string
  config_schema: string
  ai_callable: boolean
  enabled: boolean
  tags: string
  output_fields_schema?: Record<string, OutputFieldDef>
  created_at?: string
  updated_at?: string
}

export const getToolList = async () => {
  const response = await request.get<Tool[]>('/api/v1/tools')
  return response.data
}

export const getToolDetail = async (code: string) => {
  const response = await request.get<Tool>(`/api/v1/tools/${code}`)
  return response.data
}

export const describeToolOutput = async (code: string, config: Record<string, any>) => {
  const response = await request.post<Record<string, OutputFieldDef>>(
    `/api/v1/tools/${code}/describe-output`,
    config
  )
  return response.data
}

File: internal/models/tool.go
package models

import "auto-forge/pkg/utools"

// Tool 工具模型（用于API响应，不存储在数据库）
type Tool struct {
    Code               string                                `json:"code"`
    Name               string                                `json:"name"`
    Description        string                                `json:"description"`
    Category           string                                `json:"category"`
    Version            string                                `json:"version"`
    Author             string                                `json:"author"`
    Icon               string                                `json:"icon"`
    ConfigSchema       string                                `json:"config_schema"`
    AICallable         bool                                  `json:"ai_callable"`
    Enabled            bool                                  `json:"enabled"`
    Tags               string                                `json:"tags"`
    OutputFieldsSchema map[string]utools.OutputFieldDef      `json:"output_fields_schema,omitempty"`
}
